src/main/java/org/jabref/logic/ai/summarization/SummariesService.java

```diff
@@ -54,6 +61,47 @@ public SummariesService(AiPreferences aiPreferences,
         this.shutdownSignal = shutdownSignal;
         this.filePreferences = filePreferences;
         this.taskExecutor = taskExecutor;
+
+        configureDatabaseListeners(stateManager);
+    }
+
+    private void configureDatabaseListeners(StateManager stateManager) {
+        stateManager.getOpenDatabases().addListener((ListChangeListener<BibDatabaseContext>) change -> {
+            while (change.next()) {
+                if (change.wasAdded()) {
+                    change.getAddedSubList().forEach(this::configureDatabaseListeners);
+                }
+            }
+        });
+    }
+
+    private void configureDatabaseListeners(BibDatabaseContext bibDatabaseContext) {
+        // GC was eating the listeners, so we have to fall back to the event bus.
+        bibDatabaseContext.getDatabase().registerListener(new EntriesChangedListener(bibDatabaseContext));
+    }
+
+    private class EntriesChangedListener {
+        private final BibDatabaseContext bibDatabaseContext;
+
+        public EntriesChangedListener(BibDatabaseContext bibDatabaseContext) {
+            this.bibDatabaseContext = bibDatabaseContext;
+        }
+
+        @Subscribe
+        public void listen(EntriesAddedEvent e) {
+            e.getBibEntries().forEach(entry -> {
+                if (aiPreferences.getAutoGenerateSummaries()) {
+                    summarize(entry, bibDatabaseContext);
+                }
+            });
+        }
+
+        @Subscribe
+        public void listen(FieldChangedEvent e) {
+            if (e.getField() == StandardField.FILE && aiPreferences.getAutoGenerateSummaries()) {
+                summarize(e.getBibEntry(), bibDatabaseContext);
+            }
+        }
     }
```

Comment:
Move this code to UI.

The logic should not depend on the UI (see architecture tests).

Reason: AI should also be usable via CLI.

Reply:
`StateManager` belongs to UI?

Reply:
Yes. See the `package` of the StateManger.

https://github.com/JabRef/jabref/blob/b44f30e461c967fa7b2dde4aedabf9a4bd74a73c/src/main/java/org/jabref/gui/StateManager.java#L1

You can also see it, because it holds the list of opened BibTeX files

https://github.com/JabRef/jabref/blob/b44f30e461c967fa7b2dde4aedabf9a4bd74a73c/src/main/java/org/jabref/gui/StateManager.java#L59C1-L60C110